File: src/main/java/com/github/njuro/jard/config/security/captcha/CaptchaVerificationResult.java
package com.github.njuro.jard.config.security.captcha;

import java.util.List;

/**
 * Result of verification of CAPTCHA token.
 *
 * @see CaptchaProvider
 */
public interface CaptchaVerificationResult {

  /** @return true if CAPTCHA token was successfully verified, false otherwise */
  boolean isVerified();

  /** @return list of errors in case verification of token failed */
  List<String> getErrors();
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/config/security/captcha/CaptchaVerificationResult.kt
package com.github.njuro.jard.rewrite.config.security.captcha

/**
 * Result of verification of CAPTCHA token.
 *
 * @see CaptchaProvider
 */
interface CaptchaVerificationResult {
    /** @return true if CAPTCHA token was successfully verified, false otherwise
     */
    val isVerified: Boolean

    /** @return list of errors in case verification of token failed
     */
    val errors: List<String>
}
